service/http/handler.go

```diff
@@ -178,4 +178,12 @@ func (h *Handler) resolveIP(r *Request) {
 	if r.Header.Get("X-Real-Ip") != "" {
 		r.RemoteAddr = fetchIP(r.Header.Get("X-Real-Ip"))
 	}
+
+	if r.Header.Get("CF-Connecting-IP") != "" {
+		r.RemoteAddr = fetchIP(r.Header.Get("CF-Connecting-IP"))
+	}
+
+	if r.Header.Get("True-Client-IP") != "" {
+		r.RemoteAddr = fetchIP(r.Header.Get("True-Client-IP"))
+	}
```

Comment:
Generally, it's looking good to me, but, in the case of CF, `X-Real-Ip` header will be renamed. And we can distinguish those 2 parts to be a little bit efficient. Could you please add the `return` statement after `r.RemoteAddr = fetchIP(r.Header.Get("X-Real-Ip"))`? So, in the case of the CF (cold path), we did only 1 redundant operation (check for the empty `X-Real-Ip`), and in the general case, we did 1 useful operation (check for `X-Real-Ip`) and then return instead of doing 1 useful and 2 redundant operations. Thanks :)

Reply:
Done, I changed sequence of checks, "True-Client-IP" only available in CF enterprise plans and needs to enabled manually, opposite of CF-Connecting-IP, that exists for each query